subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/AbstractCrossTaskIncrementalJavaCompilationIntegrationTest.groovy

```diff
@@ -284,602 +60,45 @@ abstract class AbstractCrossTaskIncrementalJavaCompilationIntegrationTest extend
         'String'     | '"foo" + "bar"'
     }
 
-    @Unroll
-    @NotYetImplemented
-    //  Can re-enable with compiler plugins. See gradle/gradle#1474
-    def "change in an upstream class with non-private constant causes rebuild only if same constant is used and no direct dependency (#constantType)"() {
-        java api: ["class A {}", "class B { final static $constantType x = $constantValue; }"], impl: ["class X { $constantType foo() { return $constantValue; }}", "class Y {int foo() { return -2; }}"]
-        impl.snapshot { run "compileJava" }
-
-        when:
-        java api: ["class B { /* change */ }"]
-        run "impl:compileJava"
-
-        then:
-        impl.recompiledClasses('X')
-
-        where:
-        constantType | constantValue
-        'boolean'    | 'false'
-        'byte'       | '(byte) 125'
-        'short'      | '(short) 666'
-        'int'        | '55542'
-        'long'       | '5L'
-        'float'      | '6f'
-        'double'     | '7d'
-        'String'     | '"foo"'
-        'String'     | '"foo" + "bar"'
-    }
-
-    @Unroll
-    @NotYetImplemented
-    //  Can re-enable with compiler plugins. See gradle/gradle#1474
-    def "constant value change in an upstream class causes rebuild if previous constant value was used in previous build (#constantType)"() {
-        java api: ["class A {}", "class B { final static $constantType x = $constantValue; }"], impl: ["class X { $constantType foo() { return $constantValue; }}", "class Y {int foo() { return -2; }}"]
-        impl.snapshot { run "compileJava" }
+    // This test describes the current behavior, not the intended one, expressed by the test just above
+    def "changing an unused non-private constant causes full rebuild"() {
```

Comment:
Don't we need the tests with changing constants for Groovy? Since we are using a compiler plugin, those should actually work there, right?

Reply:
Seems like Groovy doesn't inline constants. Given the code 

```
class A { 
  public static final int MAGIC=1234;
}
``` 

and 

```
class B { 
  int f() { 
    return A.MAGIC; 
  } 
}
```

Java compiler inlines the result:

```
public int f();
    descriptor: ()I
    flags: (0x0001) ACC_PUBLIC
    Code:
      stack=1, locals=1, args_size=1
         0: sipush        1234
         3: ireturn
```

But Groovy compiler doesn't:

```
public int f();
    descriptor: ()I
    flags: (0x0001) ACC_PUBLIC
    Code:
      stack=2, locals=2, args_size=1
         0: invokestatic  #19                 // Method $getCallSiteArray:()[Lorg/codehaus/groovy/runtime/callsite/CallSite;
         3: astore_1
         4: aload_1
         5: ldc           #32                 // int 0
         7: aaload
         8: ldc           #34                 // class A
        10: invokeinterface #40,  2           // InterfaceMethod org/codehaus/groovy/runtime/callsite/CallSite.callGetProperty:(Ljava/lang/Object;)Ljava/lang/Object;
        15: invokestatic  #46                 // Method org/codehaus/groovy/runtime/typehandling/DefaultTypeTransformation.intUnbox:(Ljava/lang/Object;)I
        18: ireturn
        19: nop
        20: nop
        21: athrow
```

`CompileStatic` compiler compiles to:

```
public int f();
    descriptor: ()I
    flags: (0x0001) ACC_PUBLIC
    Code:
      stack=1, locals=1, args_size=1
         0: getstatic     #33                 // Field A.MAGIC:I
         3: ireturn
         4: nop
         5: nop
         6: athrow
```

So the constant tests are not applicable for Groovy incremental compilation.